coda/coda_mdstore/tests/test_views.py

```diff
@@ -591,60 +591,42 @@ def test_raises_http404_file_handle_is_falsy(self, rf):
-    """
-    Tests for coda_mdstore.views.bagURLListScrape.
-    """
-
-    @pytest.fixture(autouse=True)
-    def setup_fixtures(self, monkeypatch):
-        file_handle = mock.Mock()
-        file_handle.info().getheader.side_effect = ['text/plain', '255']
-
-        self.getFileHandle = mock.Mock(return_value=file_handle)
-        monkeypatch.setattr(
-            'coda_mdstore.views.getFileHandle', self.getFileHandle)
-
-    def test_raises_not_found_when_object_not_found(self, rf):
+    def test_response_content(self, rf):
+        """Test the response contains the url and parsed paths from the mocked file handle."""
+        self.getFileHandle.return_value.url = 'https://coda/testurl'
+        # Mock what gets read from the manifest file.
+        self.getFileHandle.return_value.readline.side_effect = [
+            b'192e635b17a9c2aea6181f0f87cab05d  data/file01.txt',
+            b'18b7c500ef8bacf7b2151f83d28e7ca1  data/file02.txt',
+            b'']
+        bag = FullBagFactory.create()
         request = rf.get('/')
-        response = views.bagURLListScrape(request, 'ark:/00001/id')
+        response = views.bagURLList(request, bag.name)
 
-        assert response.status_code == 404
-        assert isinstance(response, http.HttpResponseNotFound)
-        assert response.content == "There is no bag with id 'ark:/00001/id'."
+        assert (b'https://coda/data/file02.txt\n'
+                b'https://coda/data/file01.txt') in response.content
+        assert response.status_code == 200
 
-    @pytest.mark.xfail(reason='pairtreeCandidateList is not available '
-                              'in scope.')
-    def test_raises_http404_file_handle_is_falsy(self, rf):
-        FullBagFactory.create()
-        self.getFileHandle.return_value = False
+    def test_response_content_has_topFiles(self, rf):
+        """Test topFiles are returned in the response."""
+        self.getFileHandle.return_value.url = 'https://coda/testurl'
+        # Mock what gets read from the manifest file.
+        self.getFileHandle.return_value.readline.side_effect = [
+            b'192e635b17a9c2aea6181f0f87cab05d  data/file01.txt',
+            b'18b7c500ef8bacf7b2151f83d28e7ca1  data/file02.txt',
+            b'']
+        bag = FullBagFactory.create()
         request = rf.get('/')
-
-        with pytest.raises(http.Http404):
-            views.bagURLListScrape(request, 'ark:/00001/id')
-
-    @pytest.mark.xfail(reason='pairtreeCandidateList is not available '
-                              'in scope.')
-    def test_response_content(self):
-        assert 0
+        # Mock file names found at the bag's root level.
+        with mock.patch('coda_mdstore.views.getFileList',
+                        return_value=['bagit.txt', 'bag-info.txt']):
+            response = views.bagURLList(request, bag.name)
+
+        assert (b'https://coda/bag-info.txt\n'
+                b'https://coda/bagit.txt\n'
+                b'https://coda/data/file02.txt\n'
```

Comment:
This is really tiny, but I don't think we really need to check for the two `data/` files here, since that seems to be the point of the previous test.

Reply:
Well..I don't HAVE to test both the `data/` files, bit I did, 'cos I'm making them both as side_effect for mocked `getFileHandle` and asserting both will make it easier to read the test.

Reply:
fair enough :thumbsup: